### PR TITLE
Nodejs week3

### PR DESCRIPTION
**Pull request by nataliasowa**
Warmup exercise

### CODE REVIEW

File: nodejs/week3/warmup/calculator.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        .reduce((a, b) => Number(a) * Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+router.post("/divide", (request, response) => {
+    let result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) / Number(b));
+    response.send(`This is the result ${result}`);
+});
```
Comment (dsamourkasidis) [inline nodejs/week3/warmup/calculator.js:73]: These are post requests, but you're still accessing the query parameters, which are passed in as part of the url.
You need to read the parameters from the request body!
Here is some more information about the difference between GET and POST requests:
https://www.guru99.com/difference-get-post-http.html and https://www.postman.com/ is a useful tool to make POST requests and test your api.
Also, here is how you can access the body parameters in nodejs using express:
https://nodejs.dev/learn/get-http-request-body-data-using-nodejs
You basically need to do this:
```
app.use(
  express.urlencoded({
    extended: true
  })
)

app.use(express.json())

let result = Object.values(request.body)

File: nodejs/week3/warmup/calculator.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+// POST section
+
+router.post("/add", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) + Number(b), 0);
```
Comment (dsamourkasidis) [inline nodejs/week3/warmup/calculator.js:50]: You are using reduce which is a nice way to do this :+1: 
Also, one more thing you could do, optionally, to make your code more re-usable is to create a separate function for adding/subtracting etc.. and call them both in GET and POST requests.
That way, you avoid copy pasting the same code :)

**Review by dsamourkasidis — CHANGES_REQUESTED** [review]
Very nice work Natalia!
You kept the routes short and to the point which makes your code clean and readable :+1: 
The main thing you need to alter is that even though you've created post requests, you're still accessing the parameters from the request's url (.query). Read up a bit on that, fix it and you're good to go!

**Comment by NataliaS1994** [comment]
Thank you for the review! I will soon implement changes based on your feedback

**Comment by nataliasowa** [comment]
Thank you for the review!  changes implemented based on the feedback

**Pull request merged**